buildrunner/sshagent/__init__.py

```diff
@@ -249,15 +253,33 @@ def get_ssh_agent_image(self):
         """
         Get and/or create the image used to proxy the ssh agent to a container.
         """
+        buildrunner_config = buildrunner.config.BuildRunnerConfig.get_instance()
         if not self._ssh_agent_image:
-            self.log.write("Creating ssh-agent image\n")
-            image = legacy_builder.build_image(
-                path=SSH_AGENT_PROXY_BUILD_CONTEXT,
-                docker_registry=self.docker_registry,
-                nocache=False,
-                pull=False,
-            )
-            self._ssh_agent_image = image
+            if buildrunner_config.run_config.use_legacy_builder:
+                self.log.write("Creating ssh-agent image\n")
+                image = legacy_builder.build_image(
+                    path=SSH_AGENT_PROXY_BUILD_CONTEXT,
+                    docker_registry=self.docker_registry,
+                    nocache=False,
+                    pull=False,
+                )
+                self._ssh_agent_image = image
+            else:
+                native_platform = (
+                    self._multiplatform_image_builder.get_native_platform()
+                )
+                platforms = [native_platform]
+                built_images_info = (
+                    self._multiplatform_image_builder.build_multiple_images(
+                        platforms=platforms,
+                        path=SSH_AGENT_PROXY_BUILD_CONTEXT,
+                        cache=False,
+                        pull=False,
+                        use_threading=False,
+                    )
+                )
+                assert len(built_images_info.built_images) == 1
```

Comment:
Maybe this assertion should have an error message describing what to do if it fails (i.e. retry?)

Reply:
Honestly I prefer using an `if ... raise` pattern here instead of assertion since the assertion pretty much always throws a bad stacktrace and can be difficult to read.

Reply:
Yeah, that makes sense. I can update that.